crates/bevy_render/src/render_resource/uniform_buffer.rs

```diff
@@ -155,18 +155,45 @@ impl<T: ShaderType + WriteInto> UniformBuffer<T> {
 ///
 /// [std140 alignment/padding requirements]: https://www.w3.org/TR/WGSL/#address-spaces-uniform
 pub struct DynamicUniformBuffer<T: ShaderType> {
-    scratch: DynamicUniformBufferWrapper<Vec<u8>>,
     buffer: Option<Buffer>,
     label: Option<String>,
     changed: bool,
     buffer_usage: BufferUsages,
     _marker: PhantomData<fn() -> T>,
 }
 
+pub struct DynamicUniformBufferWriter<'a, T> {
```

Comment:
TODO: Need to document this type and it's relationship with `DynamicUniformBuffer`.